plugins/dpsn/examples/dpsn_agent.py

```diff
@@ -0,0 +1,256 @@
+    get_state_fn=get_worker_state,
+    action_space=[
+        plugin.get_function("subscribe"),
+        plugin.get_function("unsubscribe"),
+        plugin.get_function("shutdown"),
+        check_status_function,
+        complete_task_function  # Add the task completion function
+    ],
+)
+
+# Initialize the agent
+agent = Agent(
+    api_key=os.environ.get("GAME_API_KEY"),
+    name="DPSN Market Data Agent",
+    agent_goal="Monitor SOLUSDT market data from DPSN, process messages for 2 minutes, then clean up and exit.",
+    agent_description=(
+        "You are an AI agent specialized in DPSN market data processing. Follow these steps in order:\n\n"
+        
+        "1. IMMEDIATELY subscribe to this DPSN topic: 0xe14768a6d8798e4390ec4cb8a4c991202c2115a5cd7a6c0a7ababcaf93b4d2d4/SOLUSDT/ohlc\n"
+        
+        "2. Once subscribed, inform the user that you are listening for market data updates.\n"
+        
+        "3. When you receive messages on this topic:\n"
+        "   - Report that you received a message\n"
+        
+        "4. After subscribing, periodically use the check_collection_status function to check if:\n"
+        "   - 2 minutes have passed, OR\n"
+        "   - At least 3 messages have been received\n"
+        
+        "5. Only when check_collection_status indicates collection is complete:\n"
+        "   - Unsubscribe from the topic using the unsubscribe function\n"
+        "   - Shut down the DPSN client connection using the shutdown function\n"
+        
+        "6. FINAL STEP: After shutting down the DPSN connection, call the mark_task_complete function.\n"
+        "   This will finish your task and automatically exit the program after 5 seconds.\n"
+        "   This must be the VERY LAST function you call.\n"
+
+        "IMPORTANT: The program will exit automatically after you mark the task complete.\n"
+        
+        "Available functions:\n"
+        "- subscribe: Subscribe to a DPSN topic to receive data\n"
+        "- unsubscribe: Unsubscribe from a DPSN topic\n"
+        "- shutdown: Close the DPSN client connection\n"
+        "- check_collection_status: Check if collection is complete\n"
+        "- mark_task_complete: Mark the agent's task as complete and exit the program\n\n"
+        
+        "Available topic:\n"
+        "- 0xe14768a6d8798e4390ec4cb8a4c991202c2115a5cd7a6c0a7ababcaf93b4d2d4/SOLUSDT/ohlc"
+    ),
+    get_agent_state_fn=get_agent_state_fn,
+    workers=[
+        subscription_worker
+    ]
+)
+
+try:
+    print("\n=== DPSN AGENT STARTING ===")
+    print("This agent will:")
+    print("1. Subscribe to the SOLUSDT/ohlc topic")
+    print("2. Collect messages for 2 minutes or at least 3 messages")
+    print("3. Unsubscribe and clean up")
+    print("4. Exit automatically\n")
+    
+    agent.compile()
+    agent.run() 
+except Exception as e:
+    print(f"Error running agent:{e}")
+    import traceback
+    traceback.print_exc()
```

Comment:
```suggestion
    traceback.print_exc()

```

leave a blank line as of POSIX standard

Reply:
@Ang-dot  fixed in [new commit](https://github.com/game-by-virtuals/game-python/pull/110/commits/50855eeacdbe690ced60077a680b7236fc45aee2)